mountpoint-s3/src/prefetch/caching_stream.rs

```diff
@@ -80,11 +86,25 @@ where
     Client: ObjectClient + Clone + Send + Sync + 'static,
     Cache: DataCache + Send + Sync,
 {
-    fn new(client: Client, cache: Arc<Cache>, config: RequestTaskConfig) -> Self {
-        Self { client, cache, config }
+    fn new(
+        client: Client,
+        cache: Arc<Cache>,
+        backpressure_limiter: BackpressureLimiter,
+        config: RequestTaskConfig,
+    ) -> Self {
+        Self {
+            client,
+            cache,
+            backpressure_limiter,
+            config,
+        }
     }
 
-    async fn get_from_cache(self, range: RequestRange, part_queue_producer: PartQueueProducer<Client::ClientError>) {
+    async fn get_from_cache(
```

Comment:
It seems that we are changing how often this method is being called: before every time the prefetcher asked for more data (new `RequestTask`), we would first check the cache and fall back to the client at the first cache miss. Now new `RequestTask`s are only spawned on out-of-order reads, while sequential data is requested via backpressure. This means that a fully sequential read will switch entirely to the client after a single cache miss.

We should discuss whether this is acceptable and in any case document it.